Synths now relies on access permissions as well.

PsiOmegaDelta opened the pull request:
Or to a greater degree at least. Ensures station bound synthetics will not be able to utilize Syndicate/Heist equipment/vessels, as well as the opposite for potential syndicate synths.
- [x] Map access permissions on the various non-station shuttles. Awaiting map unfreeze.
- [x] Ensure buttons check access properly for synths.
- [x] Ensure syndicate computers check access properly for synths.
- [x] Give the synth syndicate module an item to acquire access rights from id cards.
- [x] Anything else I might come across as I do things.

Could also be useful if, for example, you had a server with multiple factions and their respective synths should not necessarily be able to interact with objects of another faction.
Includes bonus refactoring of borg appearance selection.

Also makes equipment respect the canremove flag where issues have been found.

PsiOmegaDelta committed:
Synths now relies on access permissions as well.

Ensures station bound synthetics will not be able to utilize Syndicate/Heist equipment/vessels, as well as the opposite for potential syndicate synths.

mwerezak commented:
Nice, this has been needed for a while I think.

Kearel commented:
It was pretty weird that merc-borgs had access.

If I recall the syndiborg has an emag, so an ID might not be necessary?

PsiOmegaDelta commented:
Sometimes you don't want to break doors and other equipment permanently. Being able to read ids like your fellow operatives would therefor be helpful.

PsiOmegaDelta committed:
Merge remote-tracking branch 'upstream/dev' into 150817-SynthAccess

PsiOmegaDelta committed:
Synth access and icon update changes.

Refactors how synthetics set their modules. On login, if appearance selection was initiated but not finalized, the selection options popup again. Fixes a long-standing issue of borgs being able to look like the standard module by loging out.
Ensures the syndicate agent IDs and syndicate borgs will always have the same initial access permissions.
Removes now unnecessary syndicate snowflake code from NanoUI.
Synths now utilize actual ids.

PsiOmegaDelta committed:
Merge remote-tracking branch 'upstream/dev' into 150817-SynthAccess

Conflicts:
	code/game/objects/items/weapons/cards_ids.dm

PsiOmegaDelta committed:
Makes it possible for syndicate borgs to copy access from other ids.

PsiOmegaDelta committed:
Changelog entries.

mwerezak commented:
Is the synth's ID checked when interacting with doors/APCs/air alarms?

PsiOmegaDelta commented:
Anything that inherits from /obj.

Kearel commented:
How will this effect ID consoles/anything else that eats ID cards.

PsiOmegaDelta commented:
I will have to do something about the syndicate borg.. the station bound synths shouldn't be handing their ids away as they are never in a hand slot, but I'll check.

mwerezak commented:
You could always create an ID scanner module that is separate from ID card inheritance and updates the robot's ID card when used on an ID.

PsiOmegaDelta commented:
It would be nicer if nodrop was actually respected.

PsiOmegaDelta committed:
Id card handling changes.

Equipment now first checks if an id card may be removed, rather than always forcefully doing so.
Station synthetics now have a custom id, rather than re-using the captain's spare, as utilizing robotic storage would otherwise make their id cards available for public use.

mwerezak commented:
:+1:
